crates/typst/src/text/font/color.rs

```diff
@@ -11,13 +11,34 @@ use crate::syntax::Span;
 use crate::text::{Font, Glyph};
 use crate::visualize::Image;
 
-/// Tells if a glyph is a color glyph or not in a given font.
+/// Returns true if a glyph is a color glyph.
+///
+/// The `sbix`, `cbdt`, `SVG`, and `COLR` tables are searched, in order, for a matching glyph.
 pub fn is_color_glyph(font: &Font, g: &Glyph) -> bool {
     let ttf = font.ttf();
     let glyph_id = GlyphId(g.id);
-    ttf.glyph_raster_image(glyph_id, 160).is_some()
-        || ttf.glyph_svg_image(glyph_id).is_some()
-        || ttf.is_color_glyph(glyph_id)
+    let pixels_per_em = 160;
+
+    // Ultimately this can still fail as we only support
+    // PNG glyphs but the sbix table can also contain JPEG
+    // and TIFF glyphs.
+    if let Some(sbix) = ttf.tables().sbix {
+        if sbix.best_strike(pixels_per_em).is_some() {
+            return true;
+        }
```

Comment:
Don't we also need to check for the glyph ID somewhere?

Reply:
Sure